client/src/components/byline-postpicker/index.jsx

```diff
@@ -0,0 +1,48 @@
+// External dependencies.
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import { useCallback } from '@wordpress/element';
+import { PostPicker } from '@alleyinteractive/block-editor-tools';
+import PropTypes from 'prop-types';
+
+function BylinePostpicker({
+  addAuthorLabel,
+  id,
+  onUpdate,
+  profilesApiUrl,
+}) {
+  const doProfileSearch = useCallback((profileId) => {
+    apiFetch({ url: addQueryArgs(profilesApiUrl, { id: profileId }) })
+      .then((rawResults) => {
+        onUpdate(rawResults[0]);
+      });
+  }, [profilesApiUrl, onUpdate]);
+
```

Comment:
Wondering if memo-izing this would make sense here? For instance,

```js
import { useMemo } from '@wordpress/element';

function BylinePostpicker({ addAuthorLabel, id, onUpdate, profilesApiUrl }) {
  // ...existing code...

  const postPickerProps = useMemo(() => ({
    allowedTypes: ['profile'],
    onUpdate: (profile) => doProfileSearch(profile)
  }), [doProfileSearch]);

  return (
    <>
      {/* ...existing code... */}
      <PostPicker {...postPickerProps} />
    </>
  );
}
```

Could prevent unnecessary re-renders if the component's parent re-renders but these props haven't changed... not 100% sure if this is needed here though, my react skills are still developing.

Reply:
This isn't a necessary optimization here - typically, we don't memoize the definition of component props - React should be smart enough to determine what needs re-render here.